src/Headers.php

```diff
@@ -58,58 +58,49 @@ class Headers implements Countable, Iterator
      * will be lazy loaded)
      *
      * @param  string $string
-     * @param  string $EOL EOL string; defaults to {@link EOL}
+     * @param  string $EOL DEPRECATED
      * @throws Exception\RuntimeException
      * @return Headers
      */
-    public static function fromString($string, $EOL = self::EOL)
+    public static function fromString($string, $EOL = null)
     {
         $headers     = new static();
         $currentLine = '';
         $emptyLine   = 0;
 
         // iterate the header lines, some might be continuations
-        $lines = explode($EOL, $string);
-        $total = count($lines);
-        for ($i = 0; $i < $total; $i += 1) {
-            $line = $lines[$i];
-
+        foreach (preg_split('/\R/', $string) as $line) {
             // Empty line indicates end of headers
             // EXCEPT if there are more lines, in which case, there's a possible error condition
             if (preg_match('/^\s*$/', $line)) {
-                $emptyLine += 1;
+                $emptyLine++;
```

Comment:
Please revert; the previous is less error-prone. Read "JavaScript: the Good Parts" for an explanation of why. (Yes, the book is on JS; however, many of the principles stated in it apply to any C-style language.)